### PR TITLE
CB-17780 Missing check for Acceptable classes with Generics

### PR DESCRIPTION
**Pull request by bergerdenes**
In CB-17642 the constructor parameters and class private fields
were not appropriately checked if they were a Generic type.
This change inspects the constructors' parameters in a different way
by parsing the type name (unfortunately there is no better way as of now).

The 2 interesting classes in this PR are `FlowPayloadSerializabilityChecker` and `GenericParameterTypeParser`.

### CODE REVIEW

**Comment by horadla23** [comment]
pls resolve conflicts

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudInstance.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public CloudInstance(@JsonProperty("instanceId") String instanceId,
             @JsonProperty("authentication") InstanceAuthentication authentication,
             @JsonProperty("subnetId") String subnetId,
             @JsonProperty("availabilityZone") String availabilityZone,
-            @JsonProperty("params") Map<String, Object> params) {
+            @JsonProperty("parameters") Map<String, Object> params) {
```
Comment (horadla23) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudInstance.java:57]: was this intentional? won't this cause any issue? do we store this anywhere in the db? if yes is this backward comaptible?

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/Group.java [hunk 2]
```diff
@@ -41,69 +43,71 @@ public class Group extends DynamicModel {
...
+            @JsonProperty("security") Security security,
+            @JsonProperty("skeleton") CloudInstance skeleton,
+            @JsonProperty("parameters") Map<String, Object> parameters,
+            @JsonProperty("instanceAuthentication") InstanceAuthentication instanceAuthentication,
+            @JsonProperty("loginUserName") String loginUserName,
+            @JsonProperty("publicKey") String publicKey,
+            @JsonProperty("rootVolumeSize") int rootVolumeSize,
+            @JsonProperty("identity") Optional<CloudFileSystemView> identity,
+            @JsonProperty("deletedInstances") List<CloudInstance> deletedInstances,
+            @JsonProperty("network") GroupNetwork groupNetwork,
```
Comment (horadla23) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/Group.java:109]: same here, the value of jsonproperty became different than name of parameter, is this backward compatible?

Comment (bergerdenes) [reply cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/Group.java:109]: it is intentional as the field name is `network`, its getter name is `public GroupNetwork getNetwork()` and therefore the Json Property name must be `network` too

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudInstance.java (hunk 1 above)
Comment (bergerdenes) [reply cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudInstance.java:57]: it is intentional as below. The field and getter name is `parameters`

Comment (horadla23) [reply cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/CloudInstance.java:57]: ok

**Review by horadla23 — APPROVED** [review]

**Pull request merged**